admin/services/service-manager/attendance/service-manager.attendance.data.service.ts

```diff
@@ -0,0 +1,65 @@
+import { TYPES } from '../../data-access/sql.service'
+const sqlService = require('../../data-access/sql.service')
+
+export class ServiceManagerAttendanceDataService {
+  static async getAttendanceCodes (): Promise<AttendanceCode[]> {
+    const sql = `
+      SELECT code, reason, [order], visible
+      FROM mtc_admin.attendanceCode
+      WHERE isPrivileged = 0
+      ORDER BY [order]`
+    return sqlService.readonlyQuery(sql)
+  }
+
+  static async setVisibility2 (attendanceCodes: AttendanceCodeVisibility[]): Promise<void> {
```

Comment:
Do we still need setVisibility() and setVisibility2()?

Reply:
good catch, will amend!